### ISSUE TITLE
Upload fails with "invalid encoding"

### ISSUE THREAD

**Issue by tobice** [dlenski/stravacli#6]
When I try to upload a TCX file (not compressed), it fails:

```
stravalib.exc.ActivityUploadFailed: [{'resource': 'upload', 'field': 'gzip', 'code': 'invalid encoding'}]
```

As a temporary workaround I managed to get it working by removing gzip compression from the Python code.

**Comment by dlenski** [comment]
Hmmm, does it no longer allow you to upload _any_ gzip-compressed file?

- Strava's official docs show [.{tcx,gpx.fit}.gz as supported options](https://developers.strava.com/docs/uploads/#how-to-upload-an-activity) as allowed 
- So does the Python `stravalib` module (https://github.com/stravalib/stravalib/blob/be5d7403b01d3d8608ddbcfbe12dad8340178fac/src/stravalib/client.py#L886)
- I can't find any other issue reporting problems with gzip'ed uploads

> As a temporary workaround I managed to get it working by removing gzip compression from the Python code.

Good to know. Can you show exactly what you changed?

XML-based file formats like TCX and GPX have a ton of redundancy in them, so compression makes a lot of sense especially when sending them over the network.

**Comment by tobice** [comment]
> Good to know. Can you show exactly what you changed?

I replaced
```
upstat = client.upload_activity(cf, ext[1:] + '.gz', title, desc, private=args.private, activity_type=args.activity_type)
```
with
```
upstat = client.upload_activity(uf, ext[1:], title, desc, private=args.private, activity_type=args.activity_type)
```

> XML-based file formats like TCX and GPX have a ton of redundancy in them, so compression makes a lot of sense especially when sending them over the network.

While that's technically true I wonder whether it's an actual problem. I'm currently uploading to Strava my whole history of about 800 TCX files and I'm constantly running into Strava's rate limiting :sweat_smile: I'm only able to upload ~20 files every 15 minutes and ~150 files per day. Interestingly enough, the documented rate limits are much higher, but that seems to only apply to non-upload requests.

The bottom line is that the network is hardly a bottleneck here.
